Find a functionally equivalent code:android.view.inputmethod.BaseInputConnection.sendKeyEvent:COMMENT
Method Modifier: public      
Comment:/**
 * Provides standard implementation for sending a key event to the window
 * attached to the input connection's view.
 */

@@ -10,14 +10,15 @@
             viewRootImpl.dispatchKeyFromIme(event);
         }
     }
+    mIMM.notifyUserAction();
     return false;
 }

